vit_pytorch/masked_prediction_loss.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class MaskedPredictionLoss(nn.Module):
+    def __init__(self,
+                 patch_size,
+                 img_size,
+                 device="cpu"):
+        super(MaskedPredictionLoss, self).__init__()
+        self.patch_size = patch_size
+        self.num_patch_axis = img_size // patch_size
+
+    def _transform_targets(self, targets, masked_patches):
+        masked_patches_dims = []
+        for masked_patch in masked_patches:
+            height_offset = self.patch_size * ((masked_patch) // self.num_patch_axis)
+            width_offset = self.patch_size * ((masked_patch) % self.num_patch_axis)
+            masked_patches_dims.append([height_offset, width_offset])
+
+        target_patches = []
+        for target in range(targets.shape[0]):
+            for masked_patch in masked_patches_dims:
+                height_offset, width_offset = masked_patch
+                extracted_patch = targets[target, :,
+                                        height_offset:height_offset + self.patch_size,
+                                        width_offset:width_offset + self.patch_size]
+                target_patches.append(extracted_patch)
+
+        target_patches_tensor = torch.stack(target_patches)
+        target_patches_tensor = (target_patches_tensor // 0.34).long()
+        encoded_targets = F.one_hot(target_patches_tensor, 3)
```

Comment:
Also a bit confused by this line

Reply:
The one hot line?

Reply:
ohh i get it, this is for the 3 bit values for each channel? do you think for discretization, perhaps https://pytorch.org/docs/stable/generated/torch.searchsorted.html would be well-suited?

Reply:
Yeah, search sorted would definitely work for that. Do you think that search sorted would perform better than the current method with one hot?

Reply:
Just asking out of curiosity btw (trying to learn how to write better code)

Reply:
yea i think so! one of the methods based on searchsorted, bucketize https://pytorch.org/docs/stable/generated/torch.bucketize.html is made just for this purpose

Reply:
it would be a lot clearer just to use it